src/telemetry.sym

```diff
@@ -15,4 +15,5 @@ TM_3_0_0 {
     tm_create_record;
     tm_set_payload;
     tm_send_record;
+    tm_set_event_id;
 } TM_2_0_0;
```

Comment:
I think you should create a new `TM_4_0_0` block, and add `tm_set_event_id` to it with global visibility. This also means you should bump the library version to account for the new symbol. I think it's `SHAREDLIB_REVISION` you'll want to bump, but I would need to read the libtool docs for a refresher.